v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/utils/CassandraDriverConfigLoader.java

```diff
@@ -0,0 +1,147 @@
+    URL url = loadSingleFile(path);
+    LOG.debug("Loaded Cassandra Driver config from path {}", path);
+    try {
+      DriverConfigLoader.fromUrl(url).getInitialConfig();
+      return DriverConfigLoader.fromUrl(url);
+    } catch (ConfigException.Parse parseException) {
+      LOG.error(
+          "Parsing error while parsing Cassandra Driver config from path {}", path, parseException);
+      throw parseException;
+    }
+  }
+
+  /**
+   * Load the Cassandra Config from a file as a {@link java.io.Serializable} {@link OptionsMap}.
+   * This {@link OptionsMap} can be stored in any object that needs to implement {@link
+   * java.io.Serializable}. At the time of opening a conneciton to Cassandra, it can be deserialized
+   * by {@link CassandraDriverConfigLoader#fromOptionsMap(OptionsMap)}. Note: Implementation Detail,
+   * Cassandra Driver does not provide a direct method to convert a link {@link DriverConfigLoader}
+   * into an {@link OptionsMap}, or build an {@link OptionsMap} from a file.
+   *
+   * @param path A complete gcs path to the config file of the form "gs://path/to/file".
+   * @return DriverConfigLoader.
+   * @throws FileNotFoundException - If file is not found at specified path.
+   */
+  public static OptionsMap loadToSerializableOptionsMap(String path) throws FileNotFoundException {
+    OptionsMap optionsMap = new OptionsMap();
+    DriverConfigLoader configLoader = load(path);
+    configLoader
+        .getInitialConfig()
+        .getProfiles()
+        .forEach(
+            (profileName, profile) ->
+                profile.entrySet().forEach(e -> putOptionToMap(optionsMap, profileName, e)));
+
+    return optionsMap;
+  }
+
+  /**
+   * Load the {@link DriverConfigLoader} from {@link java.io.Serializable} {@link OptionsMap} which
+   * was obtained as a part of {@link
+   * CassandraDriverConfigLoader#loadToSerializableOptionsMap(String)}.
+   *
+   * @param optionsMap
+   * @return DriverConfigLoader.
+   */
+  public static DriverConfigLoader fromOptionsMap(OptionsMap optionsMap) {
+    return DriverConfigLoader.fromMap(optionsMap);
+  }
+
+  @VisibleForTesting
+  protected static URL loadSingleFile(String path) throws FileNotFoundException {
+    URL[] urls = JarFileReader.saveFilesLocally(path);
+    if (urls.length == 0) {
+      LOG.error("Could not load any Cassandra driver config file from specified path {}", path);
+      throw (new FileNotFoundException("No file found in path " + path));
+    }
+    if (urls.length > 1) {
+      LOG.error(
+          "Need to provide a single Cassandra driver config file in the specified path {}. Found {} ",
+          path,
+          urls);
+      throw (new IllegalArgumentException(
+          String.format(
+              "Need to provide a single Cassandra driver config file in the specified path %s. Found %d files",
+              path, urls.length)));
+    }
+    return urls[0];
+  }
+
+  @VisibleForTesting
+  protected static void putOptionToMap(
```

Comment:
nit: Rename to putInOptionsMap or something. The current method rename sounds like the options map is created/added completely.

Reply:
Thanks.
This method actually is `putOptionToOptionMap`, which sort of becomes redundant and hence had used `putOptionToMap`.
Changed this to `putInOptionsMap` as suggested. That works too.